skimage/util/dask_compatibility.py

```diff
@@ -0,0 +1,493 @@
+from __future__ import division
+import numpy as np
+import dask.array as da
+import operator
+import collections
+import functools
+import six
+import psutil      # New dependency
+
+"""
+Provides compatibility to use Dask and NumPy arrays on a shared codebase.
+
+The goal of this module is to provide transparent compatibility functions
+with the same names, signatures, and behavior from the base NumPy (``np``)
+and Dask (``da``) namespaces. Ideally this module would be imported and used
+as follows::
+
+>>> import dask_compatibility as dc
+>>> dc.rint(arr)  # Identical behavior if arr is from Dask or NumPy
+"""
+
+__all__ = ['asanyarray',
+           'asarray',
+           'clip',
+           'dtype_conversion_inplace',
+           'eager_min',
+           'eager_max',
+           'empty',
+           'est_best_chunk',
+           'floor',
+           'maximum',
+           'rint',
+           ]
+
+
+def _graceful_compute(variable):
+    """
+    Attempts to call .compute() on every item in variable.
+    """
+    # Shortcut on NumPy arrays first
+    if isinstance(variable, np.ndarray):
+        # Return the NumPy array
+        return variable
+
+    # Dask arrays are not iterable, but NumPy arrays are.
+    elif not isinstance(variable, collections.Iterable):
+        # Could be a Dask array, try to compute it
+        try:
+            return variable.compute()
+        except AttributeError:
+            # Not a Dask array, return the variable.
+            return variable
+
+    # Might be multiple values in a container
+    else:
+        # TODO: graceful computing of items in dicts
+        try:
+            # Attempt to gracefully compute each item
+            return tuple(_graceful_compute(x) for x in variable)
```

Comment:
You want to use `da.compute` (the function) instead of calling `.compute()` multiple times. This version removes all parallelism, potentially doing a lot of extra work.